File: LeetCode-Problems/Easy/Reverse-Integer.py
# Python
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        neg = x;
        if (x < 0):
            x = -x
        num = str(x)[::-1]
        num = int(num)
        if (num < -(2**31) or num > 2**31 - 1):
            return 0
        if (neg < 0):
            return -num;
        return num;

File: LeetCode-Problems/Easy/Reverse-Integer.java
//Java
class Solution {
    public int reverse(int x) {
        long returnedInt = 0;
        while (x != 0){
            returnedInt = returnedInt * 10;
            returnedInt = returnedInt + x % 10;
            x = x/10;
            
        }
        if (returnedInt < Integer.MIN_VALUE || returnedInt > Integer.MAX_VALUE){
            return 0;
        }
        else{
            return (int)returnedInt;
        }
    }
}
